modules/core/src/smithy4s/schema/EnumTag.scala

```diff
@@ -16,28 +16,26 @@
 
 package smithy4s.schema
 
-sealed trait EnumTag[+E]
```

Comment:
This is the most meaningful change of this PR : the changes in modelling of the tag allows for the removal of the `total` function in EnumerationSchema. Additionally, the `Closed` vs `Open` distinction is now carried by an optional `unknown` value in the 2 case classes that now compose the ADT. It simplifies some pattern match.

Reply:
using a function `E => String` and `E => Int` just clicks in so nicely with all the use cases in the various visitor, wow